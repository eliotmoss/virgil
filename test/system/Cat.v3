def buf = Array<byte>.new(4096);

def main(args: Array<string>) -> int {
	for (i < args.length) {
		var fd = System.fileOpen(args[i], true);
		if (fd < 0) continue;
		while (true) {
			var r = System.read(fd, buf);
			if (r <= 0) break;
			System.write(1, buf[0 ... r]);
		}
		System.fileClose(fd);
	}
	return 0;
}
