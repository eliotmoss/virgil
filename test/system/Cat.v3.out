Test input 1234567890!@#$%^&*()
Test input 1234567890!@#$%^&*()
